bakery/keys.go

```diff
@@ -98,13 +99,15 @@ type ThirdPartyInfo struct {
 type ThirdPartyLocator interface {
 	// ThirdPartyInfo returns information on the third
 	// party at the given location. It returns ErrNotFound if no match is found.
+	// This method must be safe to call concurrently.
 	ThirdPartyInfo(ctx context.Context, loc string) (ThirdPartyInfo, error)
 }
 
 // ThirdPartyStore implements a simple ThirdPartyLocator.
 // A trailing slash on locations is ignored.
 type ThirdPartyStore struct {
-	m map[string]ThirdPartyInfo
+	mu sync.Mutex
```

Comment:
you could use a RWMutex

Reply:
I usually don't consider RWMutexes unless the mutex is likely to be highly contended (e.g. the read or the write path is slow), but why not? Done.